src/wasm/WasmModule.cpp

```diff
@@ -227,6 +227,11 @@ uint32_t WasmModule::getArgc()
     return argc;
 }
 
+std::vector<std::string> WasmModule::getArgv()
+{
+    return argv;
```

Comment:
This is the only change I need in the `WasmModule` base class. It is used in the `args_get` system call.

The reason this is not needed in the WAVM implementation is because we use `WasmModule::argv` which we call from [`WAVMWasmModule::writeArgvToMemory`](https://github.com/faasm/faasm/blob/d60d2a8d42842085bd6b5d8961a54fdcf6ee73ba/src/wavm/WAVMWasmModule.cpp#L499). IMO the approach in WAVM adds unnecessary complexity. This means WAVM and WAMR's implementation slightly differ. I am happy to change either if you think it preferrable.

Reply:
`argv` is not private, it's `protected` (so that it can be accessed directly in a subclass). In general I actually think it's better to push as much logic as we can into the `XYZWasmModule` classes, hence it's all contained in one place, not scattered over the intrinsic/ native call implementations, and is in theory more reusable.